helper/resource/error.go

```diff
@@ -0,0 +1,21 @@
+package resource
+
+type NotFoundError struct {
+	LastError    error
+	LastRequest  interface{}
+	LastResponse interface{}
+	Message      string
+	Retries      int
+}
+
+func (e *NotFoundError) Error() string {
+	if e.Message != "" {
+		return e.Message
+	}
+
+	return "couldn't find resource"
+}
+
+func NewNotFoundError(err string) *NotFoundError {
```

Comment:
> Does that make sense? Happy to discuss further!

It does, I'm still kind of internally undecided what's the best solution though. :smile:
I'd love to have the helper function, but definitely not with just a single argument.

I will remove it for now, so it's not tempting for people to use it and we can revisit all occurrences of `NotFoundError{}` after a couple of months/year to see what are the common arguments that we can require in helper(s).

i.e. I will postpone the decision so I can gain more data and be more confident about the helper interface.